src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -44,10 +44,10 @@ protected override void Initialize()
         {
             base.Initialize();
 
-            ForegroundThreadAffinitizedObject.CurrentForegroundThreadData = ForegroundThreadData.CreateDefault();
+            ForegroundThreadAffinitizedObject.CurrentForegroundThreadData = ForegroundThreadData.CreateDefault(defaultKind: CommandLineMode);
             Debug.Assert(
-                ForegroundThreadAffinitizedObject.CurrentForegroundThreadData.Kind == ForegroundThreadDataKind.Wpf ||
-                ForegroundThreadAffinitizedObject.CurrentForegroundThreadData.Kind == ForegroundThreadDataKind.JoinableTask);
+                ForegroundThreadAffinitizedObject.CurrentForegroundThreadData.Kind == Wpf ||
+                ForegroundThreadAffinitizedObject.CurrentForegroundThreadData.Kind == JoinableTask);
```

Comment:
This assert would still fire in the CommandLine case.

Perhaps change to `ForegroundThreadAffinitizedObject.CurrentForegroundThreadData.IsValid()`?